docs/guides/advanced/testing.md

```diff
@@ -0,0 +1,183 @@
+---
+title: Testing
+---
+
+# Testing your code
+
+## In the browser
+
+Packs built using the web editor in the Pack Studio can only be run and tested in a live doc. Since users will always use a released version of your Pack, you can continue to build and test new versions without affecting existing users. To configure a doc to always use the latest version of your Pack, regardless of what's released, follow these steps:
+
+1. Open the doc you want to use for testing.
+1. Open the Pack in the sidebar.
+1. Open the **Settings** tab of the Pack.
+1. Click on the **Open Pack maker tools** button.
+1. Click the gear icon (:octicons-gear-16:) to open the **Pack maker settings** dialog.
+1. In the **Installed in this doc** dropdown select **Latest Version**.
+
+We recommend building out a dedicated doc for testing, which you can use to validate that your Pack is working as expected. Use your Pack with a variety of inputs, and compare the output to an expected value.
+
+
+## On your local machine
+
+When developing locally using the [`coda` CLI][cli], you can leverage some utilities in the SDK to help you write unit tests and integration tests for your Pack. These utilities include:
+
+* Helper functions to execute a specific formula or sync from your pack definition.
+* Mock fetchers (using `sinon`) to simulate HTTP requests and responses.
+* Validation of formula inputs and return values to help catch bugs both in your test code and your formula logic.
+* Hooks to apply authentication to http requests for integration tests.
+
+You’ll find testing and development utilities in `packs-sdk/dist/development`.
+
+The primary testing utilities are `executeFormulaFromPackDef` and `executeSyncFormulaFromPackDef`. You provide the name of a formula, a reference to your Pack definition, and a parameter list, and the utility will execute the formula for you, validate the return value, and return it to you for further assertions. These utilities provide sane default execution contexts, and in the case of a sync, will execute your sync formula repeatedly for each page of results, simulating what a real Coda sync will do.
+
+By default, these utilities will use an execution environment that includes a mock fetcher that will not actually make http requests. You can pass your own mock fetcher if you wish to configure and inspect the mock requests.
```

Comment:
Ditto. This is probably my bad capitalization from the old doc :)

Reply:
Done